application-task-manager-default/src/main/java/com/xwiki/taskmanager/internal/TaskBlockProcessor.java

```diff
@@ -141,30 +142,32 @@ public Block createTaskLinkBlock(String reference, int taskNumber)
     public List<Block> generateTaskContentBlocks(String assignee, Date duedate, String text,
         SimpleDateFormat storageFormat) throws TaskException
     {
-        Map<String, String> mentionParams = new HashMap<>();
-        Map<String, String> dateParams = new HashMap<>();
-
-        mentionParams.put("style", "FULL_NAME");
-        mentionParams.put("reference", assignee);
-        mentionParams.put("anchor", assignee.replace('.', '-') + '-' + RandomStringUtils.random(5, true, false));
-
-        dateParams.put(DATE, storageFormat.format(duedate));
-
-        MacroBlock mentionBlock = new MacroBlock("mention", mentionParams, true);
-        MacroBlock dateBlock = new MacroBlock(DATE, dateParams, true);
-
         XDOM newTaskContentXDOM =
-            getTaskContentXDOM(new MacroBlock("temporaryMacro", new HashMap<>(), text, false),
+            getTaskContentXDOM(new MacroBlock("temporaryMacro", new HashMap<>(), text == null ? "" : text, false),
                 Syntax.PLAIN_1_0);
 
         Block insertionPoint = newTaskContentXDOM.getFirstBlock(new ClassBlockMatcher(ParagraphBlock.class),
             Block.Axes.DESCENDANT_OR_SELF);
         if (insertionPoint == null) {
             insertionPoint = newTaskContentXDOM;
         }
-        insertionPoint.addChild(mentionBlock);
-        insertionPoint.addChild(dateBlock);
 
+        if (assignee != null) {
+            Map<String, String> mentionParams = new HashMap<>();
+            mentionParams.put("style", "FULL_NAME");
+            mentionParams.put("reference", assignee);
+            // TODO: Possible improvement: use the IdGenerator from the XDOM of the document.
+            mentionParams.put("anchor", assignee.replace('.', '-') + '-' + RandomStringUtils.random(5, true, false));
```

Comment:
I don't think there is any code that generates the anchor, except for when the user introduces a mention by shortcut.

Reply:
You're right. I had seen this https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-12.10/xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-ui/src/main/resources/XWiki/Mentions/MentionsMacro.xml#L152-L163 but it's indeed called only when inserting the mention macro from the WYSIWYG editor. I don't understand why we don't use the same logic to compute a default value for the anchor in the macro, instead of putting this value in the wiki syntax, where you need to be careful to update it if you change the mentioned user manually.